accepted/0022-git-buildpack.md

```diff
@@ -0,0 +1,61 @@
+# Git Support
+
+## Proposal
+
+Introduce an (optional) Git buildpack into each language buildpack family. The buildpack
+will be charged with providing `git` metadata as build-time and run-time environment
+variables.
+
+## Motivation
+
+Applications of any language may have a need for information extracted from
+`git` metadata. For example, the commit sha might be used as versioning
+information when building a Go binary or used when running a Rails application.
+
+## Implementation
+
+A new `git` buildpack will be developed to detect whether the `.git` directory, and then
+read the directory to extract the following environment variables so that they
+can be included in the built image.
+
+- `GIT_COMMIT_SHA`
```

Comment:
Is it worth considering something more generic like `REVISION` or `SCM_REVISION`, so that if,  for instanc, we added a Subversion buildpack someday, downstream buildpacks wouldn't need to know which SCM is being used?